presto-main/src/main/java/com/facebook/presto/sql/planner/PlanBuilder.java

```diff
@@ -102,13 +101,10 @@ public PlanBuilder appendProjections(Iterable<Expression> expressions, SymbolAll
         }
 
         ImmutableMap.Builder<Symbol, Expression> newTranslations = ImmutableMap.builder();
-        ParameterRewriter parameterRewriter = new ParameterRewriter(parameters, getAnalysis());
```

Comment:
Nice! Did you remove all usages of `ParameterRewriter`? 
Do we still need `TranslationMap#addIntermediateMapping`? I think it was introduced to support `Parameters`

Also I wonder if this fixes: https://github.com/prestodb/presto/issues/8001